biz.aQute.bndlib/src/aQute/bnd/build/model/BndEditModel.java

```diff
@@ -676,13 +676,23 @@ public void setIncludeSources(boolean includeSources) {
 		doSetObject(aQute.bnd.osgi.Constants.SOURCES, oldValue, includeSources, defaultFalseBoolFormatter);
 	}
 
+	public boolean hasPrivatePackageInstruction() {
+		return objectProperties.containsKey(Constants.PRIVATEPACKAGE);
+	}
+
 	public List<String> getPrivatePackages() {
-		return doGetObject(aQute.bnd.osgi.Constants.PRIVATE_PACKAGE, listConverter);
+		String key = getPrivatePackageKey();
+		return doGetObject(key, listConverter);
 	}
 
 	public void setPrivatePackages(List<? extends String> packages) {
 		List<String> oldPackages = getPrivatePackages();
-		doSetObject(aQute.bnd.osgi.Constants.PRIVATE_PACKAGE, oldPackages, packages, stringListFormatter);
+		String key = getPrivatePackageKey();
+		doSetObject(key, oldPackages, packages, stringListFormatter);
+	}
+
+	private String getPrivatePackageKey() {
+		return hasPrivatePackageInstruction() ? Constants.PRIVATEPACKAGE : Constants.PRIVATE_PACKAGE;
```

Comment:
It is permissible to use both `-privatepackage` and `Private-Package`. This change seems to ignore `Private-Package` if you have also used `-privatepackage`.

There are many other places where BndEditModel is sort of broken when we have dual instructions such as `-includeresource` and `Include-Resource` :-(

Reply:
Yeah, I agree. I have modified the implementation a bit to return a merged list of entries specified in `<Private-Package, -privatepackage>` and `<Include-Resource, -includeresource>` groups. This will make the UI more consistent. The problem with #2937 will be solved since all the private packages (irrespective of where they are mentioned) will be listed on the rich editor view. However, while adding the packages from the rich editor, I add the new entries to the existing one, that is, either `Private-Package` (if present) or 
`-privatepackage`. Similarly for `Include-Resource` and `-includeresource`.